arangod/Cluster/ClusterFeature.cpp

```diff
@@ -605,20 +612,22 @@ void ClusterFeature::start() {
 
   comm.increment("Current/Version");
 
+
+
   ServerState::instance()->setState(ServerState::STATE_SERVING);
 }
 
-void ClusterFeature::beginShutdown() {}
+void ClusterFeature::beginShutdown() {
+  shutdownHeartbeatThread();
```

Comment:
This is a potentially far-reaching change, since the heartbeat thread is gone a lot earlier. Why do we have to do this? Let's discuss.